packages/cli/src/runners/task-managers/task-manager.ts

```diff
@@ -158,6 +158,14 @@ export abstract class TaskManager {
 				});
 			}
 
+			const { staticData: incomingStaticData } = resultData;
+
+			// if the runner sent back static data, it may have changed, so update it
+			if (incomingStaticData) {
+				workflow.setStaticData(incomingStaticData);
+				workflow.staticData.__dataChanged = true;
```

Comment:
Can we do this inside the `Workflow` class as it's an implementation detail of it?

Reply:
That's because the use case is different. The constructor initializes the workflow class, which makes sense that the static data hasn't changed at that point. Here we are explicitly saying "the static data changed, here is the new static data". So the operation is different: initialize static data vs set/override/change static data.